rotkehlchen/globaldb/handler.py

```diff
@@ -2129,6 +2131,99 @@ def delete_location_asset_mappings(
+                    )
+                    if skip_errors:
+                        log.error(error_msg)
+                    else:
+                        raise InputError(error_msg) from e
+
+    @staticmethod
+    def update_counterparty_asset_mappings(
+            entries: list[CounterpartyAssetMappingUpdateEntry],
+            skip_errors: bool = False,
+    ) -> None:
+        """Updates the mapped asset identifiers in the counterparty_asset_mappings table
+        based on their counterparty symbol.
+
+        May raise(if skip_errors is False):
+            - InputError if any of the pairs of counterparty and symbol does not exist
+        """
+        with GlobalDBHandler().conn.write_ctx() as cursor:
+            for entry in entries:
+                cursor.execute(
+                    'UPDATE counterparty_asset_mappings SET local_id=? WHERE counterparty=? AND symbol=?',  # noqa: E501
+                    (
+                        entry.asset.serialize(),
+                        entry.counterparty,
+                        entry.counterparty_symbol,
+                    ),
+                )
+                if cursor.rowcount != 1:
+                    error_msg = (
+                        f'Failed to update the counterparty asset mapping of {entry.counterparty_symbol} '  # noqa: E501
+                        f'for {entry.counterparty} because it does not exist in the DB.'
+                    )
+                    if skip_errors:
+                        log.error(error_msg)
+                    else:
+                        raise InputError(error_msg)
+
+    @staticmethod
+    def delete_counterparty_asset_mappings(
+            entries: list[CounterpartyAssetMappingDeleteEntry],
+            skip_errors: bool = False,
+    ) -> None:
+        """Deletes the mappings of given asset identifiers for the given counterparty from the
+        counterparty_asset_mappings table.
+
+        May raise (if skip_errors is False):
+            - InputError if any of the pairs of counterparty and symbol does not exist
+        """
+        with GlobalDBHandler().conn.write_ctx() as cursor:
+            for entry in entries:
+                cursor.execute(
+                    'DELETE FROM counterparty_asset_mappings WHERE counterparty=? AND symbol=?',
+                    (
+                        entry.counterparty,
+                        entry.counterparty_symbol,
+                    ),
+                )
+                if cursor.rowcount != 1:
+                    error_msg = (
+                        f'Failed to delete the counterparty asset mapping of {entry.counterparty_symbol} '  # noqa: E501
+                        f'for {entry.counterparty} because it does not exist in the DB.'
+                    )
+                    if skip_errors:
+                        log.error(error_msg)
+                    else:
+                        raise InputError(error_msg)
```

Comment:
the logic for this and the locations seems quite similar. What changes is the query executed, the arguments and a bit the erroor message so can you maybe:

1. Make this structure of function with the write_cusor, error handling and so on a function that takes as argument: entries, skip_error and sql query.
2. Create a methid in `CounterpartyAssetMappingDeleteEntry` and int the location equivalent `.serialize_for_db()` that creates the tuples. Here you can use seralize_for_db()[1:]
3. make a __str__ on `CounterpartyAssetMappingDeleteEntry` you sou can call it directly on the error message


Then you can just call this wrapped by passing the sql argument adjusted if it is a location or counterparty. Likewise for the other functions of update/add